rust/main/agents/scraper/src/chain_scraper/mod.rs

```diff
@@ -249,28 +269,24 @@ impl HyperlaneSqlDb {
                 .collect::<HashMap<_, _>>();
 
             for (block_ref, _) in blocks_to_insert.drain(..) {
-                block_ref.id = hashes[&block_ref.hash];
+                hashes.get(&block_ref.hash).map(|id| block_ref.id = *id);
```

Comment:
similar here, the `.map` is a bit misleading to me - I thought we were going to return a mapped version of `hashes`. I'd prefer something like this for readability
```suggestion
                if let Some(id) = hashes.get(&block_ref.hash) {
                    block_ref.id = *id;
                }
```

Reply:
Fixed.